### ISSUE TITLE
on GUIDs and rendering posts from Redmatrix 

### ISSUE THREAD

**Issue by rivendale2010** [diaspora/diaspora#5695]
usually i can access any public post with /posts/GUID   

but it does not work in case of redmatrix post having complex guid like 892dbd78f8b2ee24535f30a9791686f3b4dcca230e43fdd5d04401b165ff31c6@deadsuperhero.com

the post does exist with the full GUID in the database, but in the production log i can see that when it goes to render the post view it truncates the GUID at the period between deadsuperhero and com   

{"id"=>"892dbd78f8b2ee24535f30a9791686f3b4dcca230e43fdd5d04401b165ff31c6@deadsuperhero"} 404 Not Found   

if i URL encode https://some.pod/posts/892dbd78f8b2ee24535f30a9791686f3b4dcca230e43fdd5d04401b165ff31c6%40deadsuperhero%2Ecom the result is the same

**Comment by jhass** [comment]
This is because of Rails format parameter, you can append .json or .xml to the post to get a respective representation. To allow dots in the component one would have to disable that feature and solely depend on the accept header, which I'm not sure we're specifying everywhere we're using that route for different formats. Besides I frequently slap .json on a post for debugging :P So I would tend to have this a won't fix.

**Comment by rivendale2010** [comment]
if you are aware of it and don't think it's a problem in other areas then cool :) the logic in some of my scripts would work better if it were there but i can be fine without, i do still have post/123 which will at least work locally

**Comment by redmatrix** [comment]
Would appreciate a review and possibly re-open. This came up again in this thread:

https://deadsuperhero.com/channel/sean/?f=&mid=b682eeb3f65ef62f3455701cfa1ea2dc1138f0cfeed76ba8db14c14105f47c84@deadsuperhero.com

After a bit of investigation it turns out that all that is needed is to urlencode the message-id that goes into the /posts link. Doing so will future proof this from a number of different message-id formats which are in use on the interwebz. It also works with the .json parameter.

**Comment by redmatrix** [comment]
I see my own investigation conflicts with the findings of the initial issue report, but urlencode  worked just fine for me.

**Comment by denschub** [comment]
The point regarding the dot and Rails format parameter is still valid. I don't see a reason to reopen, but let's wait on other opinions.

**Comment by redmatrix** [comment]
In my test, 

https://nerdpol.ch/posts/e4bd06e0337bcfd749cf327418fc2aec3afa6a3e51cd76ad0302646d4ead5d70%40red%2evilarejo%2epro%2ebr

brought up the singe page view, and

https://nerdpol.ch/posts/e4bd06e0337bcfd749cf327418fc2aec3afa6a3e51cd76ad0302646d4ead5d70%40red%2evilarejo%2epro%2ebr.json 

brought up the alternate format (json)

In other words everything seems to work just fine.

**Comment by denschub** [comment]
Well, that's even better.

**Comment by jhass** [comment]
So what do we need to change then? Do we link to the GUID version of the URL anywhere?

**Comment by redmatrix** [comment]
It would be wherever  a /posts/xxxx link is created using the guid. I'm pretty certain you just need to urlencode the guid (only the guid) just before it gets inserted into that link. There was some confusion over this link - presumably linked to the timestamp of the post in a stream view, but I think this alternates between a guid and the post-id depending on whether or not you're logged in or the author is a contact or something like that. I was only able to see post-ids when I tried to duplicate the issue, but I did this while logged in and viewing a contact's "profile page".

**Comment by jhass** [comment]
I'm not aware that we generate links with guids for posts anywhere atm., hence I asked.

**Comment by redmatrix** [comment]
Apologies if not valid, as I said I wasn't able to reproduce personally. Reported to me by harryhaller from nerdpol.ch who claimed to be able to reproduce but I'm not certain and didn't ask what D* version was being used.

**Comment by jhass** [comment]
It's popular to do manually among diaspora users as a "maybe we have luck" way of linking a post in a way that it opens on the local installation, but it's not officially supported at the moment.
